### PR TITLE
Add support for interval list or bed file to Prepare Step (within limits) [VS-1164]

### PR DESCRIPTION
**Pull request by rsasch**
Sample run with interval of 5k lines: https://app.terra.bio/#workspaces/gvs-dev/RSA%20-%20GVS%20Quickstart%20V2%20/job_history/ab37dff7-12a6-46cb-834e-ffe31797caca
Sample run with interval of 10k lines (it determined that this was too many intervals and just ran for all): https://app.terra.bio/#workspaces/gvs-dev/RSA%20-%20GVS%20Quickstart%20V2%20/job_history/f150ff61-0926-4b2e-a12e-fc4e4f4f7381

### CODE REVIEW

File: scripts/variantstore/wdl/extract/create_ranges_cohort_extract_data_table.py [hunk 1]
```diff
@@ -180,7 +194,11 @@ def create_final_extract_ref_table(fq_destination_table_ref_data, enable_extract
     query_return = utils.execute_with_retry(client, "create final export ref table", sql)
     JOBS.append({'job': query_return['job'], 'label': query_return['label']})
 
-def populate_final_extract_table_with_ref(fq_ranges_dataset, fq_destination_table_data, sample_ids, use_compressed_references):
+def populate_final_extract_table_with_ref(fq_ranges_dataset, fq_destination_table_data, sample_ids, use_compressed_references, interval_list):
+    # split file into files with x lines and then run
```
Comment (mcovarr) [inline scripts/variantstore/wdl/extract/create_ranges_cohort_extract_data_table.py:198]: comment would probably make more sense below line 200 than here

File: scripts/variantstore/wdl/extract/create_ranges_cohort_extract_data_table.py [hunk 2]
```diff
@@ -214,15 +232,19 @@ def get_compressed_ref_subselect(fq_ref_table, samples, id):
...
             print(f"{fq_ref_table} query is {utils.utf8len(sql) / (1024 * 1024)} MB in length")
             query_return = utils.execute_with_retry(client, "populate destination table with reference data", sql)
             JOBS.append({'job': query_return['job'], 'label': query_return['label']})
     return
 
 
-def populate_final_extract_table_with_vet(fq_ranges_dataset, fq_destination_table_data, sample_ids):
+def populate_final_extract_table_with_vet(fq_ranges_dataset, fq_destination_table_data, sample_ids, interval_list):
+    location_string = ""
+    if interval_list:
```
Comment (mcovarr) [inline scripts/variantstore/wdl/extract/create_ranges_cohort_extract_data_table.py:245]: this one did not get a comment...

File: scripts/variantstore/wdl/extract/create_ranges_cohort_extract_data_table.py [hunk 3]
```diff
@@ -131,6 +134,17 @@ def create_extract_samples_table(control_samples, fq_destination_table_samples,
...
 
 
+def get_location_filters_from_interval_list(interval_list):
+    interval_test = pybedtools.BedTool(interval_list)
+    location_clause_list = [f"""(location >= {CHROM_MAP[interval.chrom]}{'0' * (12 - len(str(interval.start)))}{interval.start} 
+            AND location <= {CHROM_MAP[interval.chrom]}{'0' * (12 - len(str(interval.end)))}{interval.end})"""
+                            for interval in interval_test]
+    if len(location_clause_list) > 5000:
+        print(f"\n\nTrying to query over the limit of 5,000 locations; {interval_list} will be discarded, and all locations will be queried.\n\n")
+        return ""
```
Comment (mcovarr) [inline scripts/variantstore/wdl/extract/create_ranges_cohort_extract_data_table.py:144]: Shouldn't this test happen before the potentially expensive forcomp starting on line 139 (substituting `len(interval_test)` for `len(location_clause_list)`?

**Review by mcovarr — CHANGES_REQUESTED** [review]
nits

**Review by koncheto-broad — APPROVED** [review]
LGTM.  Miguel has some good suggestions, but I also think it's viable as is

**Review by mcovarr — APPROVED** [review]

**Comment by rsasch** [comment]
Test runs after PR feedback-based changes:

 - Sample run with interval of 5k lines: https://app.terra.bio/#workspaces/gvs-dev/RSA%20-%20GVS%20Quickstart%20V2%20/job_history/8ac48d8a-f2ca-4cef-bc10-271d3503d607
 - Sample run with interval of 10k lines (it determined that this was too many intervals and just ran for all): https://app.terra.bio/#workspaces/gvs-dev/RSA%20-%20GVS%20Quickstart%20V2%20/job_history/b6189b51-fa16-46c2-953c-4571045eae34

**Pull request merged**